src/core/directives/google-map-data-layer.ts

```diff
@@ -0,0 +1,67 @@
+import {Directive, EventEmitter, OnDestroy, OnInit} from '@angular/core';
+import {Subscription} from 'rxjs/Subscription';
+
+import {DataMouseEvent} from './../services/google-maps-types';
+import {DataLayerManager} from './../services/managers/data-layer-manager';
+
+let layerId = 0;
+
+@Directive({
+  selector: 'sebm-google-map-data-layer',
+  inputs: ['geoJson', 'styleFunc'],
+  outputs: ['layerClick']
+})
+export class SebmGoogleMapDataLayer implements OnInit, OnDestroy {
+  private _addedToManager: boolean = false;
+  private _id: string = (layerId++).toString();
+  private _subscriptions: Subscription[] = [];
+
+  /**
+   * This event is fired when a feature in the layer is clicked.
+   */
+  layerClick: EventEmitter<DataMouseEvent> = new EventEmitter<DataMouseEvent>();
+
+  /**
+   * The geoJson to be displayed
+   */
+  geoJson: Object|null = null;
```

Comment:
when these values can change over time, we have to handle it in ngOnChange. Please check out the other directives.